backend/cdApp.py

```diff
@@ -230,45 +211,44 @@ def post(self, course_id, operation):
             if cursor.rowcount == 0:
                 raise InvalidUsage('Entry not found in database. Please use `setpiazza` instead')
             return jsonify(message='Successfully updated piazza ID for course', course_id=course_id)
+
         else:
             raise InvalidUsage('Unknown operation type')
 
     def get(self, course_id, operation):
-        if operation != 'getpiazza':
-            raise InvalidUsage('Unknown operation type')
+        if operation == 'getpiazza':
+            cursor = g.db.execute('''SELECT piazza_cid
+                                     FROM courses
+                                     WHERE course_id=(?)''',
+                                  [int(course_id)])
+            piazza_id_str = cursor.fetchone()
+            if piazza_id_str is None:
+                raise InvalidUsage('Given course does not have a Piazza ID')
+            else:
+                return jsonify(message='Returning piazza ID for course', course_id=course_id, piazza_cid=piazza_id_str)
 
-        cursor = g.db.execute('''SELECT piazza_cid
-                                 FROM courses
-                                 WHERE course_id=(?)''',
-                              [int(course_id)])
-        piazza_id_str = cursor.fetchone()
-        if piazza_id_str is None:
-            raise InvalidUsage('Given course does not have a Piazza ID')
+        elif operation == 'getpiazzaposts':
+            cursor = g.db.execute('''SELECT piazza_cid
+                                     FROM courses
+                                     WHERE course_id=(?)''',
+                                  [int(course_id)])
+            piazza_id_str = cursor.fetchone()['piazza_cid']
+            if piazza_id_str is None:
+                raise InvalidUsage('Given course does not have a Piazza ID')
+            else: #TODO: handle errors
+                print piazza_id_str
+                p = Piazza()
+                p.user_login(email="sakekasi@ucla.edu", password="password")
+                cls = p.network(piazza_id_str)
+
+                def getPosts():
+                    for post in cls.iter_all_posts():
+                        yield json.dumps(post)
```

Comment:
the get_posts() function is a generator function. It returns an iterator which executes until a 'yield' statement, after which it stops execution until 'next' is called. this generator is used to implement streaming in flask. This is so that we don't get a ridiculous wait time on the fetching of items